### PR TITLE
add 3D tests

### PR DESCRIPTION
**Pull request by JoshuaLampert**
This duplicates the existing 2D tests to 3D in order to test `p8est`.

### CODE REVIEW

File: test/tests_basic.jl [hunk 1]
```diff
@@ -277,10 +278,225 @@ end
...
+
+
+# This belongs to the testset "p8est_qcoord_to_vertex" below.
+function iter_volume_for_p8est_qcoord_to_vertex(info::Ptr{p8est_iter_volume_info_t}, user_data)
+  info = unsafe_load(info)
+  p4est = unsafe_load(info.p4est)
+  quad = unsafe_load(info.quad)
+
+  vxyz = Array{Float64}(undef, 3)
+  p8est_qcoord_to_vertex(p4est.connectivity, info.treeid, quad.x, quad.y, quad.z, vxyz)
```
Comment (ranocha) [inline test/tests_basic.jl:291]: To follow our own guidelines, this should be

```suggestion
  info_obj = unsafe_load(info)
  p4est = unsafe_load(info_obj.p4est)
  quad = unsafe_load(info_obj.quad)

  vxyz = Array{Float64}(undef, 3)
  p8est_qcoord_to_vertex(p4est.connectivity, info_obj.treeid, quad.x, quad.y, quad.z, vxyz)
```
Could you please also adapt also the 2D version?

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks! I just have a minor remark.

File: test/tests_basic.jl (hunk 1 above)
Comment (JoshuaLampert) [reply test/tests_basic.jl:291]: Yes, I'll do that.

**Review by ranocha — APPROVED** [review]
Thanks!

**Pull request merged**